### PR TITLE
ADEN-6694 Put InstartLogic recovery behind Wikia AdBlock Detection

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

**Comment by kvas-damian** [comment]
Why do we aim for IL only?
I'm under impression that we did something wrong here:
https://github.com/Wikia/app/blob/79c7c4ab72bf17d0d099f1b6f8399a3e75dbf974/extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getPageFairBootstrap.mustache#L7-L7
https://github.com/Wikia/app/blob/3e6b69ae198cb5f1cb6ea45b3e5eb8488391168d/extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache#L7-L7

**Comment by dmnsobczak** [comment]
Added PageFair. Disabled code include if WAD will not detect AdBlock.

**Comment by kvas-damian** [comment]
Please make sure unit tests are passing. Probably merging dev is enough this time.

File: extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
-	], function (adContext) {
+	], function (doc, adContext) {
 		var context = adContext.getContext();
 
+
 		if (context.opts.instartLogicRecovery) {
-			{{{code}}}
+			if (context.opts.babRecovery) {
+				document.addEventListener('bab.blocking', function () {
+					{{{code}}}
```
Comment (kvas-damian) [inline extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache]: this code block is pretty huge. Can we somehow avoid loading it two times?

File: extensions/wikia/AdEngine/js/slot/service/srcProvider.js [hunk 2]
```diff
@@ -35,7 +45,7 @@ define('ext.wikia.adEngine.slot.service.srcProvider',  [
 		if (adContext.get('opts.premiumOnly') && !adContext.get('opts.isAdTestWiki')) {
 			originalSrc = 'premium';
 		}
-		if (isRecoverableByPF(extra) || isRecoverableByIL()) {
+		if (isRecoveryAllowedByBab(extra) && (isRecoverableByPF(extra) || isRecoverableByIL())) {
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/slot/service/srcProvider.js]: I believe you don't need to add additional check here. `instartLogic.isBlocking()` won't return false when it's not enabled. The same should be valid for PF.

Comment (dmnsobczak) [reply extensions/wikia/AdEngine/js/slot/service/srcProvider.js]: That is true actually. Removed.

File: extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache (hunk 1 above)
Comment (dmnsobczak) [reply extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache]: Only one working solution I found out: applied.

File: extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache [hunk 3]
```diff
@@ -5,14 +5,20 @@
 	], function (doc, adContext) {
 		var context = adContext.getContext();
 
-
 		if (context.opts.instartLogicRecovery) {
+			var event = doc.createEvent('Event');
+			event.initEvent('rcv.ilb', true, false);
+
+			doc.addEventListener('rcv.ilb', function () {
```
Comment (kvas-damian) [inline extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache]: Can we simplify this solution by simply defining a function and calling it when required?

File: extensions/wikia/AdEngine/js/slot/service/srcProvider.js [hunk 4]
```diff
@@ -3,11 +3,13 @@ define('ext.wikia.adEngine.slot.service.srcProvider',  [
 	'ext.wikia.adEngine.adContext',
 	'ext.wikia.aRecoveryEngine.adBlockDetection',
 	'ext.wikia.aRecoveryEngine.adBlockRecovery',
+	'wikia.window',
 	require.optional('ext.wikia.aRecoveryEngine.instartLogic.recovery')
 ], function (
 	adContext,
 	adBlockDetection,
 	adBlockRecovery,
+	win,
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/slot/service/srcProvider.js]: `win` is unused

File: extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js [hunk 5]
```diff
@@ -22,6 +22,7 @@ define('ext.wikia.adEngine.provider.factory.wikiaGpt', [
 			sraEnabled: extra.sraEnabled,
 			isInstartLogicRecoverable: extra.isInstartLogicRecoverable ? extra.isInstartLogicRecoverable(slotName) : false,
 			isPageFairRecoverable: extra.isPageFairRecoverable ? extra.isPageFairRecoverable(slotName) : false,
+			isRecoveryBehindBab: extra.isRecoveryBehindBab,
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js]: it is unused. lets remove it

File: extensions/wikia/AdEngine/js/slot/service/srcProvider.js (hunk 4 above)
Comment (dmnsobczak) [reply extensions/wikia/AdEngine/js/slot/service/srcProvider.js]: Removed.

File: extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js (hunk 5 above)
Comment (dmnsobczak) [reply extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js]: Removed too.

File: extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache (hunk 3 above)
Comment (dmnsobczak) [reply extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache]: I had some kind of brain malfunction so I did this "the weird way". Refactored.

File: extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache [hunk 6]
```diff
@@ -5,20 +5,17 @@
...
-			});
+		function runILCode() {
+			{{{code}}}
+		}
 
+		if (context.opts.instartLogicRecovery) {
 			if (context.opts.babRecovery) {
 				doc.addEventListener('bab.blocking', function () {
-					doc.dispatchEvent(event);
+					runILCode();
```
Comment (kvas-damian) [inline extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache]: do we need to wrap it in a function?

File: extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache [hunk 7]
```diff
@@ -1,11 +1,22 @@
...
 
-		if (context.opts.instartLogicRecovery) {
+		function runILCode() {
 			{{{code}}}
 		}
+
+		if (context.opts.instartLogicRecovery) {
+			if (context.opts.babRecovery) {
+				doc.addEventListener('bab.blocking', function () {
+					runILCode();
```
Comment (kvas-damian) [inline extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache]: do we need to wrap it into a function?

File: extensions/wikia/AdEngine/js/spec/slot/service/srcProvider.spec.js [hunk 8]
```diff
@@ -129,7 +129,7 @@ describe('ext.wikia.adEngine.slot.service.srcProvider', function () {
 
 	it('doesn\'t set src=premium if article isn\'t premium', function () {
 		mockContext({'opts.premiumOnly': false});
-		expect(getModule().get('abc')).not.toBe('premium');
+		expect(getModule().get('abc', {})).not.toBe('premium');
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/spec/slot/service/srcProvider.spec.js]: Are these changes required? I don't see any changes related to `srcProvider` now ;)

File: extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache (hunk 7 above)
Comment (dmnsobczak) [reply extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache]: Oh, yeah, right. Done.

File: extensions/wikia/AdEngine/js/spec/slot/service/srcProvider.spec.js (hunk 8 above)
Comment (dmnsobczak) [reply extensions/wikia/AdEngine/js/spec/slot/service/srcProvider.spec.js]: :> They're not necessary right now, reverted.

**Review by kvas-damian — APPROVED** [review]
